scenarios/waymo_motion/waymo_utility.py

```diff
@@ -0,0 +1,2202 @@
+    mng.resize(1000, 1000)
+    plt.title(f"Scenario {scenario_id}")
+
+    # Plot map
+    if scenario_dict[scenario_id][1] is None:
+        scenario_dict[scenario_id][1] = get_map_features_for_scenario(scenario)
+    plot_map_features(scenario_dict[scenario_id][1], [])
+    all_handles = get_map_handles()
+
+    if animate:
+        # Plot Trajectories
+        if scenario_dict[scenario_id][2] is None:
+            scenario_dict[scenario_id][2] = get_trajectory_data(scenario)
+        data, points, max_len, _ = plot_trajectories(
+            scenario_dict[scenario_id][2],
+            scenario_dict[scenario_id][0].objects_of_interest,
+            [],
+        )
+        all_handles.extend(get_trajectory_handles())
+        plt.legend(handles=all_handles)
+
+        def update(i):
+            drawn_pts = []
+            for (xs, ys), point in zip(data, points):
+                if i < len(xs):
+                    point.set_data(xs[i], ys[i])
+                    drawn_pts.append(point)
+            return drawn_pts
+
+        # Set Animation
+        anim = FuncAnimation(
+            fig, update, frames=range(1, max_len), blit=True, interval=100
+        )
+        out_path = os.path.join(
+            os.path.abspath(target_path), f"scenario-{scenario_id}.mp4"
+        )
+        anim.save(out_path, writer=FFMpegWriter(fps=15))
+
+    else:
+        plt.legend(handles=all_handles)
+        out_path = os.path.join(
+            os.path.abspath(target_path), f"scenario-{scenario_id}.png"
+        )
+        fig = plt.gcf()
+        fig.set_size_inches(1000 / 100, 1000 / 100)
+        fig.savefig(out_path, dpi=100)
+
+    print(f"Saving {out_path}")
+    plt.close("all")
+    return True
+
+
+def dump_plots(target_base_path: str, scenario_dict, animate=False, filter_tags=None):
+    try:
+        os.makedirs(os.path.abspath(target_base_path))
+        print(f"Created directory {target_base_path}")
+    except FileExistsError:
+        pass
+    except (OSError, RuntimeError):
+        print(f"{target_base_path} is an invalid path. Please enter a valid path")
+        return
+
+    plot_parameters = product(
+        [scenario_id for scenario_id in scenario_dict],
+        [target_base_path],
+        [scenario_dict],
+        [animate],
+        [filter_tags],
+    )
+    with Pool(min(cpu_count(), len(scenario_dict))) as pool:
+        plots_dumped = pool.starmap(save_plot, plot_parameters)
+
+    if any(plots_dumped):
+        print(f"All images or recordings saved at {target_base_path}")
+    else:
+        print(f"No images or recordings saved as no tags matched")
+
+
+def filter_scenario(scenario_tags, imported_tags, filter_tags) -> bool:
+    tags, filter_display = filter_tags
+    if filter_display == 1:
+        return all(x in scenario_tags for x in tags)
+    elif filter_display == 2:
+        return all(x in imported_tags for x in tags)
+    else:
+        return all(x in imported_tags + scenario_tags for x in tags)
```

Comment:
This can be simplified.
```suggestion
    return all(x in imported_tags + scenario_tags for x in tags)
```